include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -132,22 +310,59 @@ struct walk_n<_ExecutionPolicy, oneapi::dpl::__internal::__no_op>
 // walk_adjacent_difference
 //------------------------------------------------------------------------
 
-template <typename _ExecutionPolicy, typename _F>
-struct walk_adjacent_difference
+template <typename _ExecutionPolicy, typename _F, typename _Range1, typename _Range2>
+struct walk_adjacent_difference : public walk_vector_or_scalar_base<_Range1, _Range2>
 {
+  private:
+    using __base_t = walk_vector_or_scalar_base<_Range1, _Range2>;
     _F __f;
+    std::size_t __n;
+    oneapi::dpl::__internal::__pstl_assign __assigner;
 
-    template <typename _ItemId, typename _Acc1, typename _Acc2>
+  public:
+    walk_adjacent_difference(_F __f, std::size_t __n) : __f(std::move(__f)), __n(__n) {}
+
+    template <typename _IsFull, typename _ItemId>
     void
-    operator()(const _ItemId __idx, const _Acc1& _acc_src, _Acc2& _acc_dst) const
+    __scalar_path_impl(_IsFull, const _ItemId __idx, const _Range1 __rng1, _Range2 __rng2) const
     {
-        using ::std::get;
-
         // just copy an element if it is the first one
         if (__idx == 0)
-            _acc_dst[__idx] = _acc_src[__idx];
+            __assigner(__rng1[__idx], __rng2[__idx]);
         else
-            __f(_acc_src[__idx + (-1)], _acc_src[__idx], _acc_dst[__idx]);
+            __f(__rng1[__idx + (-1)], __rng1[__idx], __rng2[__idx]);
+    }
+    template <typename _IsFull, typename _ItemId>
+    void
+    __vector_path_impl(_IsFull __is_full, const _ItemId __idx, const _Range1 __rng1, _Range2 __rng2) const
+    {
+        using _ValueType = oneapi::dpl::__internal::__value_t<_Range1>;
+        _ValueType __rng1_vector[__base_t::__preferred_vector_size + 1];
+        // 1. Establish a vector of __preferred_vector_size + 1 where a scalar load is performed on the first element
+        // followed by a vector load of the specified length.
+        if (__idx != 0)
```

Comment:
As we could  observe, a ternary operator may provide a faster binary code...
I don't insist, but I would try to write:
`__assigner(__idx != 0 ? __rng1[__idx - 1] :__rng1[0],  __rng1_vector[0]);`

Reply:
I think it may make sense but better to check performance and then make decision how to write this code better.

Reply:
Performance is the same from my testing. However, it is simply, so I adopted it.